lib/active_fedora/file.rb

```diff
@@ -63,7 +63,15 @@ def ldp_connection
     # If this file has a parent with ldp#contains, we know it is not new.
     # By tracking exists we prevent an unnecessary HEAD request.
     def new_record?
-      !@exists && ldp_source.new?
+      !@exists && ldp_new_record?
```

Comment:
I wonder if we should be tracking `@destroyed` like core does? (https://github.com/projecthydra/active_fedora/blob/7ed4ca1cb4443b6b39dccea4dba18442809cb028/lib/active_fedora/persistence.rb#L49) Can you show me where `target.delete(record)` is calling `==`?